### DISCUSSION TITLE
Man page should mention if --cookie-jar overwrites

### DISCUSSION THREAD

**Discussion by jidanni** [curl/curl#11642] [General]
$ man curl
> -c, --cookie‐jar <filename>
>              (HTTP) Specify to which file you want curl to write all cookies after a completed operation. Curl writes all cookies from its in‐memory cookie storage to the given file...

That's great, but the man page should also say if the file is
overwritten, or if instead cookies are added and removed from it --
simply updated.

Note I am not asking a question. I'm saying the man page should mention it.

**Comment by bagder** [comment]
How is "writes all cookies from its in‐memory cookie storage to the given file" not saying it will "overwrite" the file? Can you write a file without "overwriting" it?

**Reply by jidanni** [reply]
Perhaps because e.g., browsers and many other CLI tools carefully update their cookies files,
rather than throwing all previous cookies out each subsequent run.

So I would make it clear that curl guarantees to ignore any previous cookies, even if not expired yet.

**Reply by bagder** [reply]
> curl guarantees to ignore any previous cookies, even if not expired yet.

What "previous cookies" does it ignore?

**Reply by jidanni** [reply]
curl --cookie‐jar cookies.txt ...
curl --cookie‐jar cookies.txt ...

Above, the second invocation will ignore the previously existing cookies  that are left behind from the first invocation.

This is in contrast with all other programs that I know of.

I'm not saying it's good. I'm not saying it's bad. All I'm saying is
you might want to make a note of it on the man page.

**Reply by bagder** [reply]
Is that not what the man page for `--cookie-jar` already says?

First it states that this file is for **writing** cookies to at the end:

> Specify to which file you want curl to write all cookies after a completed operation. Curl writes all cookies from its in-memory cookie storage to the given file at the end of operations.

Then it says that if there are no cookies, it does not save any:

> If no cookies are known, no data will be written.

**Reply by jidanni** [reply]
Yes but it is not clear if it doesn't read old cookies in or not.

I mean that's what all the other programs do. So it wouldn't hurt to say that it doesn't. Even though in a court of law, yes, you didn't say that you did.

But people are just worried that you forgot to say that you didn't. So I would just say it explicitly. Otherwise it's not clear if your program acts like all the other programs, which it doesn't.

**Reply by bagder** [reply]
It only talks about writing cookies so it would be weird to think that it actually reads cookies, but okay. I'll try to add something.

**Reply by bagder** [reply]
See #11661

**Comment by bagder** [comment]
> Note I am not asking a question. I'm saying the man page should mention it.

Now when  we reached a conclusion and a fix, I want to mention something.

I find your "demanding attitude" in this post (and in previous communication) quite rude. We are all in this project on equal terms. We all try to make it the best possible thing we can. No one can demand anything of anyone else.

Can I please ask you to think of yourself as one of the team in future correspondence? How do you think **we** should improve things in the future, and ideally try to propose solutions and ways forward when you find problems.

Thanks!
